### ISSUE TITLE
[ENHANCEMENT] Remove --hex-salt and add hashlist/cracker specific attack commands

### ISSUE THREAD

**Issue by zyronix** [hashtopolis/server#900]
With the most recent bug #892 and having a discussion about this, I would like to discuss the following. The hexsalt feature of hashtopolis is something that should be removed. This should be added to a task, or a hashcat specific option that should be linked to the hashlist.

Currently it is not clear what the implications of this option exactly is. And secondly, it is very hashcat specific.

**Comment by s3inlc** [comment]
The purpose of this feature was that a user has to set one time that a hashlist is having a hex salt. Otherwise the user needs to keep in mind that every time he creates a task, he needs to add the --hex-salt flag, otherwise either the task will not start at all, or even worse, it will run, but never be able to find the password.
What would your proposal of "hashcat specific option that should be linked to the hashlist" mean exactly? Would you prefer of having a free-text field on hashlist creation, to let the user add any (i.e. hashcat) specific command line arguments which should be used in any attack on this hashlist? This way it would not be hashcat specific only, but allow for any cracker specific additions to be selected.

**Comment by zyronix** [comment]
> What would your proposal of "hashcat specific option that should be linked to the hashlist" mean exactly? Would you prefer of having a free-text field on hashlist creation, to let the user add any (i.e. hashcat) specific command line arguments which should be used in any attack on this hashlist?
Exactly, having a free-text field for the attack-command at hashlist creation. You could argue that the option is specific per cracker type. Example: hashcat needs --hex-salt, but my generic-cracker does not. So this attack-command is cracker specific.

More importantly, the 'free-text' field should be displayed both at task edit screen and task detailed screen; so you know what expactly is being sent to the agent.

So maybe this feature should be two features:

 - Remove hexsalt and create 'hashlist specific attack command'
 - Resulting attack command display

**Comment by s3inlc** [comment]
I would propose we open a separate issue for the attack command display feature and keep this one for the removal of the hexsalt and adding the hashlist specific attack command option.

**Comment by zyronix** [comment]
Done #901
